lms/djangoapps/instructor_task/tests/test_base.py

```diff
@@ -350,7 +351,9 @@ def verify_rows_in_csv(self, expected_rows, file_index=0, verify_order=True, ign
         report_store = ReportStore.from_config(config_name='GRADES_DOWNLOAD')
         report_csv_filename = report_store.links_for(self.course.id)[file_index][0]
         report_path = report_store.path_to(self.course.id, report_csv_filename)
-        with report_store.storage.open(report_path) as csv_file:
+        with report_store.storage.open(report_path, 'r+b') as csv_file:
+            # Removing unicode signature (BOM) from the beginning
+            self.remove_bom_inplace(csv_file)
```

Comment:
I don't think you should remove the BOM this way.  It modifies the file on-disk, which seems like a waste.  Instead, wrap your open file in this class:
```
class SkipBom(object):
    def __init__(self, f):
        self.f = f

    def __iter__(self):
        first = next(self.f)
        if first.startswith(codecs.BOM_UTF8):
            first = first[len(codecs.BOM_UTF8):]
        yield first
        for line in self.f:
            yield line
```
This will iterate lines just like the original file, but if the first line starts with a BOM, will remove it.